inference-engine/src/vpu/graph_transformer/src/stages/reduce.cpp

```diff
@@ -127,41 +127,43 @@ class ReduceStage final : public StageNode {
 }  // namespace
 
 void FrontEnd::parseReduce(const Model& model, const ie::CNNLayerPtr& _layer, const DataVector& inputs, const DataVector& outputs) const {
-    auto layer = std::dynamic_pointer_cast<ie::ReduceLayer>(_layer);
-    VPU_THROW_UNLESS(layer != nullptr,
-                     "Layer {} of type {} is nullptr",
-                     layer->name, layer->type);
+    VPU_THROW_UNLESS(_layer != nullptr,
+                     "parseReduce expects valid CNNLayerPtr, actually got nullptr");
```

Comment:
Is this check necessary? I think check after dynamic cast is enough.
Why `_layer`, not `layer`? If it is just an artifact from previous version - could you please rename if possible?